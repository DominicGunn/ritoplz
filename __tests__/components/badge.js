'use strict'

/* eslint-env jest */

import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import Badge from '../../components/ui/badge'

describe('ui badge component', () => {
  it('should have class primary', () => {
    const wrapper = shallow(<Badge type="primary">Button</Badge>)
    expect(wrapper.hasClass('primary')).toEqual(true)
  })

  it('should have class success', () => {
    const wrapper = shallow(<Badge type="success">Button</Badge>)
    expect(wrapper.hasClass('success')).toEqual(true)
  })

  it('should have class danger', () => {
    const wrapper = shallow(<Badge type="danger">Button</Badge>)
    expect(wrapper.hasClass('danger')).toEqual(true)
  })

  it('should have children props passed', () => {
    const wrapper = shallow(<Badge>danger button</Badge>)
    expect(wrapper.find('.primary').props().children[0]).toEqual(
      'danger button'
    )
  })

  describe('snapshots', () => {
    it('should render primary badge with only children props', () => {
      const tree = renderer.create(<Badge>Badge</Badge>).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('should render success badge', () => {
      const tree = renderer.create(<Badge type="success">Badge</Badge>).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('should render danger badge', () => {
      const tree = renderer.create(<Badge type="danger">Badge</Badge>).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
