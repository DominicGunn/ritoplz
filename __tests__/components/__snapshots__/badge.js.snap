// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui badge component snapshots should render danger badge 1`] = `
<div
  className="danger"
  data-jsx={1669289405}
>
  Badge
</div>
`;

exports[`ui badge component snapshots should render primary badge with only children props 1`] = `
<div
  className="primary"
  data-jsx={1669289405}
>
  Badge
</div>
`;

exports[`ui badge component snapshots should render success badge 1`] = `
<div
  className="success"
  data-jsx={1669289405}
>
  Badge
</div>
`;
